### PR TITLE
feat: update collector with options

### PR DESCRIPTION
**Pull request by ferreiratiago**
## Description

This pull request introduces the parameter options to collector and the new method collect error.

This is to allow the collect to be created with the minium values.

## Related to

Closes https://github.com/reviewpad/reviewpad/issues/554

## Type of change

Improvements (non-breaking change without functionality)

## How was this tested?

Run `task check -f`

## Checklist

- [x] I have performed a self-review of my code
- [x] I have ran `task check -f` and have no issues

## Code review and merge strategy (ship/show/ask) 

 
 
- [x] Ask: this pull request requires a code review before merge

### CODE REVIEW

File: collector/collector.go [hunk 1]
```diff
@@ -11,60 +11,101 @@ import (
...
-	// Allows to identify the url
+	// Optional properties that are added to every event.
+	Optional *OptionalProperties
+}
+
+type OptionalProperties struct {
+	// The repository url where the events are being collected.
 	Url string
+	// Delivery ID of the event.
+	DeliveryId string
```
Comment (marcelosousa) [inline collector/collector.go:42]: What's the difference between the `DeliveryId` and the `DistinctId`?

File: collector/collector.go [hunk 2]
```diff
@@ -11,60 +11,101 @@ import (
...
-	Id        string
-	Token     string
+	Client mixpanel.Mixpanel
+	// Unique identifier that is connected to every event.
+	// It is used to identify the user.
+	DistinctId string
+	// Token to authenticate the requests and identify the project.
+	Token string
+	// Runner environment where the event is being collected.
+	// (i.e. github app, github action, playground )
```
Comment (marcelosousa) [inline collector/collector.go:25]:
```suggestion
	// (i.e. github app, github action, playground)
```

**Review by marcelosousa — APPROVED** [review]
I made a couple of comments in the diff. It's not easy to fully understand these changes and how to apply them in the consumers. It would be good to see the PR in the action repository that will use this new consumer. 
Looking at https://github.com/reviewpad/reviewpad/issues/554, it could be nice for future documentation purpose to list a couple of examples of where the `url` is not required; I thought this was always the case so that we can keep track of the entire journey. 
I think I'm missing something here but IMO it doesn't deserve blocking this PR so I'm approving the PR!

File: collector/collector.go (hunk 1 above)
Comment (ferreiratiago) [reply collector/collector.go:42]: `DistinctId` it is used to identify the user.
`DeliveryId` it is used to identify the event within the runner. For instance, in both github action and app it corresponds to the "X-GitHub-Delivery".

**Pull request merged**